scripts/west_commands/runners/nrfjprog.py

```diff
@@ -113,8 +114,46 @@ def do_run(self, command, **kwargs):
                              format(self.hex_) +
                              'Try enabling CONFIG_BUILD_OUTPUT_HEX.')
 
+        if self.family == 'NRF53':
+            # self.hex_ can contain code for both application core and network
+            # core. Code being flashed to the network core needs to have a
+            # specific argument provided to nrfjprog. If network code is found,
+            # generate two new hex files, one for each core, and flash them
+            # with the correct '--coprocessor' argument.
+            ih = IntelHex(self.hex_)
+            net_hex = IntelHex()
+            app_hex = IntelHex()
+            for s in ih.segments():
+                if s[0] >= 0x01000000:
```

Comment:
I'm assuming this is a flash address that's specific to the network core.

If so, this magic number should come from devicetree (probably from a flash reg property's value), using the python edtlib library to extract the information from the build directory's devicetree. Check out scripts/west_commands/sign.py for an example of how to use and call into edtlib.

Doing that would also probably let you find the application core addresses, which would let you guard the `else` code with a check to make sure that only code intended for the app core ends up in app_hex, in case that is valuable.